File: go/pairs/solution_test.go
package solution

import (
	"reflect"
	"testing"
)

func TestPairs(t *testing.T) {
	testCases := []struct {
		name     string
		elements []string
		want     [][]string
	}{
		{
			name:     "test case one",
			elements: []string{"a", "b", "c"},
			want: [][]string{
				{"a", "b"},
				{"a", "c"},
				{"b", "c"},
			},
		},
		{
			name:     "test case two",
			elements: []string{"a", "b", "c", "d"},
			want: [][]string{
				{"a", "b"},
				{"a", "c"},
				{"a", "d"},
				{"b", "c"},
				{"b", "d"},
				{"c", "d"},
			},
		},
		{
			name:     "test_case_three",
			elements: []string{"cherry", "cranberry", "banana", "blueberry", "lime", "papaya"},
			want: [][]string{
				{"cherry", "cranberry"},
				{"cherry", "banana"},
				{"cherry", "blueberry"},
				{"cherry", "lime"},
				{"cherry", "papaya"},
				{"cranberry", "banana"},
				{"cranberry", "blueberry"},
				{"cranberry", "lime"},
				{"cranberry", "papaya"},
				{"banana", "blueberry"},
				{"banana", "lime"},
				{"banana", "papaya"},
				{"blueberry", "lime"},
				{"blueberry", "papaya"},
				{"lime", "papaya"},
			},
		},
	}

	for _, tc := range testCases {
		t.Run(tc.name, func(t *testing.T) {
			got := pairs(tc.elements)
			if !reflect.DeepEqual(got, tc.want) {
				t.Errorf("got %v, want %v", got, tc.want)
			}
		})
	}
}

File: py/pairs/solution_test.py
import unittest

from solution import Solution


class TestSolution(unittest.TestCase):
    def setUp(self):
        self.solution = Solution()

    def test_case_one(self):
        elements = ["a", "b", "c"]
        expected = [
            ["a", "b"],
            ["a", "c"],
            ["b", "c"]
        ]
        result = self.solution.pairs(elements)
        self.assertListEqual(expected, result)

    def test_case_two(self):
        elements = ["a", "b", "c", "d"]
        expected = [
            ["a", "b"],
            ["a", "c"],
            ["a", "d"],
            ["b", "c"],
            ["b", "d"],
            ["c", "d"]
        ]
        result = self.solution.pairs(elements)
        self.assertListEqual(expected, result)

    def test_case_three(self):
        elements = ["cherry", "cranberry", "banana", "blueberry", "lime", "papaya"]
        expected = [
            ["cherry", "cranberry"],
            ["cherry", "banana"],
            ["cherry", "blueberry"],
            ["cherry", "lime"],
            ["cherry", "papaya"],
            ["cranberry", "banana"],
            ["cranberry", "blueberry"],
            ["cranberry", "lime"],
            ["cranberry", "papaya"],
            ["banana", "blueberry"],
            ["banana", "lime"],
            ["banana", "papaya"],
            ["blueberry", "lime"],
            ["blueberry", "papaya"],
            ["lime", "papaya"]
        ]
        result = self.solution.pairs(elements)
        self.assertListEqual(expected, result)
